Expliciter le choix de l’algorithme de chiffrement FranceConnect

francois-ferrandis opened the pull request:
Ceci est une copie exacte de la PR #5772 mais pour FranceConnect.

En synthèse : 
- on explicite dans le code qu'on utilise toujours `ES256`
- on évite d'utiliser `jwks.first`, c'est à dire de prendre la première valeur d'une liste qui provient d'une API, ça paraît plus robuste

J'ai observé `france_connect_v2_config.jwks` sur les API FranceConnect de sanbox et de prod, et dans les deux cas la réponse est : 

```ruby
[
  {
  "kty" => "EC",
  "use" => "sig",
  "kid" => "pkcs11:ES256:hsm",
  "alg" => "ES256",
  "crv" => "P-256",
  "x" => "[CACHÉ]",
  "y" => "[CACHÉ]"
  },
  {
  "kty" => "RSA",
  "use" => "sig",
  "kid" => "[CACHÉ]",
  "alg" => "RS256",
  "e" => "AQAB",
  "n" => "[CACHÉ]"
  }
]

# (oui, c'est des données publiques, mais je préfère être prudent et cacher les clés de chiffrement)
```

On voit bien que `ES256` arrive en premier ce qui explque que `jwks.first` fonctionne actuellement.

francois-ferrandis committed:
Expliciter le choix de l’algorithme de chiffrement FranceConnect

victormours commented on app/services/france_connect_v2_open_id_client/callback.rb:
```diff
@@ -1,5 +1,7 @@
 # voir https://github.com/france-connect/Documentation-AgentConnect/blob/main/doc_fs/technique_fca/endpoints.md
 module FranceConnectV2OpenIdClient
+  ALGORITHM = "ES256".freeze
```
On pourrait être un tout petit peu plus explicite comme ça ?
```suggestion
  ES256_ALGORITHM = "ES256".freeze
```

francois-ferrandis replied:
En effet c'est mieux !

victormours reviewed (approved):
Merci, et trop bien d'avoir ajouté la vérification au démarrage

francois-ferrandis committed:
ALGORITHM -> ES256_ALGORITHM

francois-ferrandis committed:
Ne vérifie par la présence du support au lancement du serveur

Nous avons eu des crash inexpliqués sur ce genre de code
côté ProConnect, et ça ne semble pas critique (on sera
informé par ailleurs si la méthode de chiffrement change).

francois-ferrandis committed:
Keep it simple
